src/library.cpp

```diff
@@ -157,16 +157,20 @@ void Library::loadMonitorDir(QString monitorDir)
     QSet<QString> oldDir = QSet<QString>::fromList(oldDirEntries);
     QStringList addedZims = (newDir - oldDir).values();
     QStringList removedZims = (oldDir - newDir).values();
-    setMonitorDirZims(newDir.values());
     auto manipulator = LibraryManipulator(this);
     auto manager = kiwix::Manager(&manipulator);
+    bool needsRefresh = false;
     for (auto book : addedZims) {
-        manager.addBookFromPath(book.toStdString());
+        needsRefresh = manager.addBookFromPath(book.toStdString()) ? true : needsRefresh;
```

Comment:
This is more about style but this could be rewrite with:
`needsRefresh = needsRefresh || manager.addBookFromPath(book.toStdString())` or `needsRefresh |= manager.addBookFromPath(book.toStdString())`

It will avoid a test and a branch and it is a syntax have seen more often for this use case of updating a boolean.

Reply:
Neat trick, done